Model/Product/Builder.php

```diff
@@ -174,6 +180,25 @@ public function buildImageUrl(Product $product, StoreInterface $store)
     }
 
     /**
+     * Adds the alternative image urls
+     *
+     * @param Product $product the product model.
+     * @return array
+     */
+    protected function buildAlternativeImages(Product $product)
+    {
+        $images = [];
+        $this->galleryReadHandler->execute($product);
+        foreach ($product->getMediaGalleryImages() as $image) {
```

Comment:
I've now handled this is a saner way by modifying the SDK to handle the deduplication. This is platform agnostic and works just fine. https://github.com/Nosto/nosto-php-sdk/pull/46/commits/6c7bf1f4b6bb9c023e7c98f436dc4dc65e46a4ff